controllers/imagerepository_controller.go

```diff
@@ -432,3 +444,46 @@ func (r *ImageRepositoryReconciler) recordSuspension(ctx context.Context, imageR
 		r.MetricsRecorder.RecordSuspend(*objRef, imageRepo.Spec.Suspend)
 	}
 }
+
+func parseAuthMap(config dockerConfig) (map[string]authn.AuthConfig, error) {
+	auth := map[string]authn.AuthConfig{}
+	for host, entry := range config.Auths {
+		u, err := parseURL(host)
+		if err != nil {
+			return nil, err
+		}
+
+		host = u.Host
+
+		auth[host] = entry
+	}
+
+	return auth, nil
+}
+
+func parseURL(urlStr string) (*url.URL, error) {
```

Comment:
I would reverse the logic here so that `url.Parse()` is only called once. Something like this:
```
if !strings.HasPrefix(urlStr, "http://") || !strings.HasPrefix(urlStr, "https://") {
  urlStr = fmt.Sprintf("https://%s/", urlStr)
}
u, err := url.Parse(urlStr)
```
Checking the format of the string first (if it contains the scheme) and parsing only after prepending the scheme (if it is necessary).